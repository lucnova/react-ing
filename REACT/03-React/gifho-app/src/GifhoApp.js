import React from 'react';
import PropTypes from 'prop-types';

const GifhoApp = (props) => {
	return (
		<>
			<h2>Gifho!</h2>
			<hr />
		</>
	);
};

GifhoApp.propTypes = {};

export default GifhoApp;
