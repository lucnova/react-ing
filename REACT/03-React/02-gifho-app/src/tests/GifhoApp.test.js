import { shallow } from 'enzyme';
import GifhoApp from '../GifhoApp';

/* No es posible evaluar un hook useState en Jest */

describe('testing on GifhoApp', () => {
	test('should match snapshot', () => {
		const wrapper = shallow(<GifhoApp />);

		expect(wrapper).toMatchSnapshot();
	});

	test('should display NO GifGrid components', () => {
		const wrapper = shallow(<GifhoApp defaultCategories={[]} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifCategoryGrid').length).toBe(0);
	});

	test('should display SAME LENGTH of GifGrid components', () => {
		const categoriesSent = ['Pixel Art', 'Vinesauce', 'DDLC'];
		const wrapper = shallow(<GifhoApp defaultCategories={categoriesSent} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifCategoryGrid').length).toBe(categoriesSent.length);
	});
});
