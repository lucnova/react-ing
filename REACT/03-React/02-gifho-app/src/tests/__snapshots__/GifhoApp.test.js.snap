// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on GifhoApp should display NO GifGrid components 1`] = `
<Fragment>
  <div>
    <h2>
      Gifho!
    </h2>
    <small>
      lucnova
    </small>
    <hr />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <div>
    <ol />
  </div>
</Fragment>
`;

exports[`testing on GifhoApp should display SAME LENGTH of GifGrid components 1`] = `
<Fragment>
  <div>
    <h2>
      Gifho!
    </h2>
    <small>
      lucnova
    </small>
    <hr />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <div>
    <ol>
      <GifCategoryGrid
        categoryName="Pixel Art"
        key="Pixel Art"
      />
      <GifCategoryGrid
        categoryName="Vinesauce"
        key="Vinesauce"
      />
      <GifCategoryGrid
        categoryName="DDLC"
        key="DDLC"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`testing on GifhoApp should match snapshot 1`] = `
<Fragment>
  <div>
    <h2>
      Gifho!
    </h2>
    <small>
      lucnova
    </small>
    <hr />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <div>
    <ol />
  </div>
</Fragment>
`;
